### PR TITLE
feat: created endpoint to fetch best vendors for best vendors page

### PR DESCRIPTION
**Pull request by anetta999**
## Describe your changes
- created best-vendors controller
- created getBestVendors service
- created get-best-vendors dto
- updated documentation

## Issue ticket code (and/or) and link
- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-72)
  
### **General**
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Backend
- [x] Swagger documentation updated
- [x] Database requests are optimized and not redundant
- [ ] Unit tests written
- [x] use ConfigService instead of process.env
- [ ] use transactions if there is a call chain that mutates data in different tables
- [ ] use @index decorator for frequently requested data

### CODE REVIEW

File: src/modules/users/users.service.ts [hunk 1]
```diff
@@ -797,4 +798,32 @@ export class UsersService {
...
+
+      return bestVendors.map((vendor) => ({
+        vendorId: vendor.id,
+        vendorName: vendor.name,
+        photoUrl: vendor.photoUrl,
+        products: vendor.products.map((product) => ({
+          ...product,
+          vendor: product.user,
+        })),
+      }));
```
Comment (lyudad) [inline src/modules/users/users.service.ts]: мне это не нравится у тебя сложность O(n)2 (в квадрте) что не очень хорошо если не прям очень нужно. я не понимаю зачем это делать, но если оставлять так .то нужно пагинацию добавлять, потому что же с ростом масива скорость выполнения будет очень быстро уменьшатся

Comment (anetta999) [reply src/modules/users/users.service.ts]: fixed in https://github.com/ZenBit-Tech/code-lions_be/pull/21/commits/682ce459f56d408eb7f25b76c58eb87b3e29a8bb

Comment (lyudad) [reply src/modules/users/users.service.ts]: why not here?

Comment (anetta999) [reply src/modules/users/users.service.ts]: fixed here https://github.com/ZenBit-Tech/code-lions_be/pull/21/commits/171dc3e426aea3f255257a306d7ed27d9345936b

Comment (lyudad) [reply src/modules/users/users.service.ts]: но у тебя опять map в нутри map ничего не пофиксилось, сложность где то такая же

Comment (lyudad) [reply src/modules/users/users.service.ts]: и почему у тебя slice не в выборке а в map. тебе нужно модернизировать запрос

Comment (anetta999) [reply src/modules/users/users.service.ts]: changed request logic in https://github.com/ZenBit-Tech/code-lions_be/pull/21/commits/385f5e517bbcfd0cd2fbb842cb9cd97fb7edc949

**Review by lyudad — APPROVED** [review]

**Pull request merged**